examples/30_extended/fetch_runtimes_tutorial.py

```diff
@@ -0,0 +1,281 @@
+
+#####################################################################
+# Case 3: Running and benchmarking HPO algorithms with their runtimes
+# *******************************************************************
+# We shall now optimize a similar RandomForest model for the same task using
+# scikit-learn HPO support by using GridSearchCV to optimize our earlier
+# RandomForest model's parameter `n_estimators`. scikit-learn also provides a
+# `refit_time_` for such HPO models, i.e., the time incurred by training
+# and evaluating the model on the best found parameter setting. OpenML
+# includes this in the `wall_clock_time_millis_training` measure recorded.
+
+clf = RandomForestClassifier(n_estimators=10, n_jobs=2)
+
+# GridSearchCV model
+n_iter = 5
+grid_pipe = GridSearchCV(
+    estimator=clf,
+    param_grid={"n_estimators": np.linspace(start=1, stop=50, num=n_iter).astype(int).tolist()},
+    cv=2,
+    n_jobs=2,
+)
+
+run4 = openml.runs.run_model_on_task(model=grid_pipe, task=task, n_jobs=2)
+measures = run4.fold_evaluations
+print_compare_runtimes(measures)
+print()
+
+##################################################################################
+# Like any optimisation problem, scikit-learn's HPO estimators also generate
+# a sequence of configurations which are evaluated, using which the best found
+# configuration is tracked throughout the trace.
+# The OpenML run object stores these traces as OpenMLRunTrace objects accessible
+# using keys of the pattern (repeat, fold, iterations). Here `fold` implies the
+# outer-cross validation fold as obtained from the task data splits available in OpenML.
+# Since we are running `GridSearchCV` here which itself performs a `2-fold` cross validation,
+# the runtime recorded per repeat-per fold in the run object is for the entire `fit()`
+# procedure of GridSearchCV thus subsuming the runtimes of the 2-fold CV search performed.
+
+# We earlier extracted the number of repeats and folds for this task:
+print("# repeats: {}\n# fold: {}".format(n_repeats, n_folds))
+
+# To extract the training runtime of the first repeat, first fold:
+print(run4.fold_evaluations["wall_clock_time_millis_training"][0][0])
+
+##################################################################################
+# To extract the training runtime of the 1-st repeat, 4-th (outer) fold and also
+# to fetch the parameters and performance of the evaluations made during
+# the 1-st repeat, 4-th fold evaluation by the Grid Search model.
+
+_repeat = 0
+_fold = 3
+print(
+    "Total runtime for repeat {}'s fold {}: {:4f} ms".format(
+        _repeat, _fold, run4.fold_evaluations["wall_clock_time_millis_training"][_repeat][_fold]
+    )
+)
+for i in range(n_iter):
+    key = (_repeat, _fold, i)
+    r = run4.trace.trace_iterations[key]
+    print(
+        "n_estimators: {:>2} - score: {:.3f}".format(
+            r.parameters["parameter_n_estimators"], r.evaluation
```

Comment:
Do we also have the fit time for the individual model or only the score?

Reply:
No, I wanted to ask you the same 😅 

However, to answer you, nope! I couldn't find any aspect of the runtime being captured by the `OpenMLRunTrace` class definition.

Reply:
Alright, just added this request to https://github.com/openml/OpenML/issues/413